### PR TITLE
Style OSIDB-2005: UI Space optimization

### PR DESCRIPTION
**Pull request by C-Valen**
# OSIDB-2005 Remove extra whitespace

## Checklist:

- [x] Linting passed
- [x] Type checks passed
- [x] Tests suite passed
- [x] Commits (not yet) consolidated
- [x] Changelog updated
- [x] Test cases added/updated
- [x] Jira ticket updated

## Summary:

Remove whitespace on OSIM's UI make a better use of the space available in every view aiming for a compact style.

## Changes:

- Make flaw form larger width
- Adjust form elements for a compact inline style
- Optimize space on text area fields section (descriptions)
- Make text area fields always visible (and remove buttons)
- Sightly adjust spacing on flaw attributions
- Minor UI corrections for affects sections
- Minor spacing adjustment between affect items
- Sightly adjust spacing on comments section

## Considerations:

- Depends on [OSIM-288](https://github.com/RedHatProductSecurity/osim/pull/288)
- Also asked for feedback on the demo to users
- I considered toggling text area descriptions is not required anymore as with new layout the space is already compact
- Considered to make the embargoed highlighted border cover all the sections. Reason is users may want to have feedback that they are on a embargoed flaw in different sections (e.g. adding comments)

## Demo:

image

image

Video (some things might be outdated)

https://github.com/RedHatProductSecurity/osim/assets/29104825/b94d0a1a-2988-4c51-91dc-d03175a51843

### CODE REVIEW

**Comment by superbuggy** [comment]
Love the way this looks so far

File: src/components/CvssCalculator.vue [hunk 1]
```diff
@@ -249,7 +252,7 @@ function handlePaste(e: ClipboardEvent) {
     .vector-input {
       height: 100%;
       border-radius: 0;
-      padding-left: 30px;
+      padding-left: 12px;
```
Comment (superbuggy) [inline src/components/CvssCalculator.vue:255]:
```suggestion
      padding-left: 0.75rem;
```

File: src/components/CvssCalculator.vue [hunk 2]
```diff
@@ -230,7 +230,10 @@ function handlePaste(e: ClipboardEvent) {
 .osim-input {
   display: inline-flex;
   width: 100%;
-  margin-bottom: 15px;
+
+  .form-label {
+    width: 230px;
```
Comment (superbuggy) [inline src/components/CvssCalculator.vue:235]: I'd potentially recommend using `ch` units here

File: src/components/FlawForm.vue [hunk 3]
```diff
@@ -555,11 +506,13 @@ form.osim-flaw-form :deep(*) {
   font-family: 'Red Hat Mono', monospace;
 
   .osim-flaw-form-section{
```
Comment (superbuggy) [inline src/components/FlawForm.vue:508]: Do we need `position: relative` here?

**Comment by MrMarble** [comment]
Is this ready to be reviewed? is still on Draft

**Comment by MrMarble** [comment]
image
On larger screens the labels get stretched to fill the space, I think we should control the max size for the components or maybe keep the borders for very large screens?
I can imagine some of the engineers using ultra-wide screens

File: src/components/CvssCalculator.vue (hunk 1 above)
Comment (MrMarble) [reply src/components/CvssCalculator.vue:255]: Do you know this values from memory or is there any converter? 😆

**Comment by C-Valen** [comment]
> image On larger screens the labels get stretched to fill the space, I think we should control the max size for the components or maybe keep the borders for very large screens? I can imagine some of the engineers using ultra-wide screens

I think we can just fix the label sizes for now, and leave the responsive design for a different task. 
I also think that with the prod release and multiple users getting into the application we will have a direct feedback to know if there are engineers using different resolutions to address and which we should consider. Wdyt?

**Review by superbuggy — APPROVED** [review]

**Review by MrMarble — APPROVED** [review]

**Review by tboyetteredhat — APPROVED** [review]

**Pull request merged**